### ISSUE TITLE
receiver_open event is fired even if the subscription is not found.

### ISSUE THREAD

**Issue by aliosss** [amqp/rhea#151]
I'm using this library with Azure service bus topics and subscriptions. I'm trying to find a way to ensure that my receiver link is established successfully. Currently, I'm using receiver_open, however, this event is fired even if I try to connect to a non-existing subscription. Of course it is followed by receiver_error and receiver_close events. 

My question is whether this is an intended behavior? If so, what do you suggest as the best way to check whether a receiver link is successfully open?

**Comment by grs** [comment]
Yes, that is intended, as it mirrors the protocol. You can check that the source address echoed by the peer matches what you requested. If the source address not defined or is null, then it means the subscription was not successful and will be followed immediately by a close. The address set by the peer can be obtained through context.receiver.source.address

**Comment by aliosss** [comment]
> Yes, that is intended, as it mirrors the protocol. You can check that the source address echoed by the peer matches what you requested. If the source address not defined or is null, then it means the subscription was not successful and will be followed immediately by a close. The address set by the peer can be obtained through context.receiver.source.address

Thank you for your response. Yesterday, I ran into a nasty state with Azure, where I called open_receiver, it was not followed by a close or an error and the link was not open. Unfortunately, I was not debugging, so I do not have debug logs and it is very difficult to reproduce (most probably it happened during an update at Azure side).

Now, I'm trying to detect that scenario and handle properly, therefore looking for a reliable way of detecting an open receiver link. Not sure whether context.receiver.source.address will resolve my issue but I'll try and see whether I'll run into the same issue again.
